go/exocom/src/connection/service.go

```diff
@@ -0,0 +1,73 @@
+package connection
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/Originate/exocom/go/structs"
+	"github.com/Originate/exocom/go/utils"
+	"github.com/gorilla/websocket"
+	"github.com/pkg/errors"
+)
+
+// Service manages a single websocket connection
+type Service struct {
+	manager *Manager
+	role    string
+	socket  *websocket.Conn
+}
+
+// BootstrapService creates a new Service which will alert the manager when the
+// service registers (by sending its first message) and deregisters (by disconnecting)
+func BootstrapService(manager *Manager, socket *websocket.Conn) {
+	service := Service{
+		manager: manager,
+		socket:  socket,
+	}
+	fmt.Println("bootstraping service")
+	go service.listen()
+}
+
+// Send sends the given message on the websocket
+func (s *Service) Send(message structs.Message) error {
+	serializedBytes, err := json.Marshal(message)
+	if err != nil {
+		return err
+	}
+	return s.socket.WriteMessage(websocket.TextMessage, serializedBytes)
+}
+
+// Helpers
+
+func (s *Service) listen() {
+	utils.ListenForMessages(s.socket, func(message structs.Message) error {
+		fmt.Println("received message", message)
+		if message.Name == "exocom.register-service" {
+			var err error
+			s.role, err = parseRegisterMessagePayload(message)
+			if err == nil {
+				fmt.Println("registering service")
+				s.manager.registerService(s)
+			} else {
+				s.manager.logError(err)
+			}
+		} else {
+			s.manager.onMessage(message)
+		}
+		return nil
+	}, func(err error) {
```

Comment:
Sounds good. This is an existing helper that is used in a few other places. I will update it to use channels in a separate PR